backend/udev.c

```diff
@@ -143,21 +138,16 @@ static int udev_event(int fd, uint32_t mask, void *data) {
 	}
 
 	const char *action = udev_device_get_action(dev);
-	const char *path = udev_device_get_devnode(dev);
 
 	wlr_log(L_DEBUG, "udev event for %s (%s)",
 			udev_device_get_sysname(dev), action);
 
-	if (!path || strcmp(path, udev->drm_path) != 0) {
-		goto out;
-	}
-
 	if (!action || strcmp(action, "change") != 0) {
 		goto out;
 	}
 
-	// TODO: Specify the GPU that's being invalidated
-	wl_signal_emit(&udev->invalidate_drm, udev);
+	dev_t devnum = udev_device_get_devnum(dev);
+	wl_signal_emit(&udev->invalidate_drm, &devnum);
```

Comment:
We definitely need to provide people with access to the wlr_udev instance here. You should make a new struct for the event that contains both the dev_t and the wlr_udev.